conn.go

```diff
@@ -118,12 +118,12 @@ func (c *Conn) Streams() []*Stream {
 // Close closes this connection
 func (c *Conn) Close() error {
 	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
 	if c.closed {
+		c.closeLock.Unlock()
```

Comment:
Ok, call it c.GoClose() and have it spin a goroutine if it does need to
close (i.e. do honor the "go"). in practice it wont happen

On Wed, Dec 2, 2015 at 12:49 PM Jeromy Johnson <notifications@github.com>
wrote:

> In conn.go
> <https://github.com/jbenet/go-peerstream/pull/21#discussion_r46473117>:
>
> >  	if c.closed {
> > +		c.closeLock.Unlock()
>
> thats also something we don't want though, we would end up with tons of
> random goroutines hanging around for an indeterminate period of time.
>
> Are you opposed to a method that calls close, and exits early if its being
> closed already like this one? maybe StartClosing or something?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jbenet/go-peerstream/pull/21/files#r46473117>.
>